### DISCUSSION TITLE
How about using Vue or Nuxt apps with Keycloakify?

### DISCUSSION THREAD

**Discussion by cdecinkoKnight** [keycloakify/keycloakify#540] [Ideas]
We do all of our development in Vue 3 and Nuxt 3. Any chance there could some day be a version that would allow us to use that to create custom Keycloak themes?

**Comment by garronej** [comment]
Hello @cdecinkoKnight,  
Yes in theory. There's been a POC: https://github.com/keycloak/keycloak/discussions/9093#discussioncomment-8720943  

However it's not on the short therm roadmap.

**Comment by AndreyYolkin** [comment]
It's been a long time, but now I decided to burn some tokens left and create a PoC: https://github.com/AndreyYolkin/keycloakify-vue

The login.ftl page is implemented end-to-end. It renders in vite dev on a mock kcContext and compiles to a real Keycloak theme .jar through the standard keycloakify CLI. The generated login/login.ftl references the Vue bundle via ${xKeycloakify.resourcesPath} and injects window.kcContext, exactly like a React theme.

The approach mirrors [@keycloakify/svelte](https://github.com/keycloakify/keycloakify-svelte): it depends on keycloakify as a peer dependency and reuses the framework-agnostic layers unchanged, such as KcContext types, the i18n/noJsx engine, getUserProfileApi, kcClsx, kcSanitize, the vite-plugin and the build/jar pipeline. Only the components and a thin glue layer (composables and i18n to VNode bridge) are reimplemented in Vue. The core CLI is detected via a minimal _keycloakify-custom-handler that emits a Vue-like kc.gen.ts. No changes to keycloakify core were needed.

**Reply by garronej** [reply]
Sorry for the very late reply. Let's get in toch on Discord.  

DM me `@garronej` 

https://discord.gg/mJdYJSdcm4
